### PR TITLE
Constexpr math conversions for double rad/deg conversions

### PR DESCRIPTION
**Pull request by Ryanf55**
While working on debugging the GSOF floating point exceptions, I could not debug the macros in GDB. In a quick test, switching to constexpr changed that, so I added constexpr double support for the radians to degrees conversions. 

I also did some research [here](https://sourceware.org/gdb/onlinedocs/gdb/Macros.html): `You may need to compile your program specially to provide GDB with information about preprocessor macros. Most compilers do not include macros in their debugging information, even when you compile with the -g flag.` 

Given the two options of modifying the build system or using constexpr, and the additional benefits of constexpr in terms of compile-time safety compared to macros, I chose to fix this issue by making  deg/rad conversions constexpr. 

See the picture at the end
https://github.com/ArduPilot/ardupilot/pull/23843#issuecomment-1558598144

Since constexpr provides compile-time length checking, this provides a clear advantage. I followed the given pattern for where to put the conversion functions, and replaced usage of the macro ratio with the new function, which is similar to the current float implementation. 

Based on recommendations in other PR's against overloading operations like these due to potential for expensive and implicit widening, I also named the function differently.

### CODE REVIEW

**Comment by andyp1per** [comment]
Nice!

**Comment by magicrub** [comment]
```const``` and ```constexpr``` all the things!

**Comment by khancyr** [comment]
+1 for const and constexpr !

Issue on math functions is that they aren't constexpr on the cpp norm, only gcc provide a const version, which is a pain..

**Comment by Ryanf55** [comment]
> +1 for const and constexpr !
> 
> Issue on math functions is that they aren't constexpr on the cpp norm, only gcc provide a const version, which is a pain..

Ok let's make it easy. I'll just paste in a double precision version of pi instead of trying to be fancy. We can solve the constexpr math later.

**Comment by Ryanf55** [comment]
Looks like this PR introduced some unit test failures, where values are now off by a small amount. It's not clear why yet. I'll look into it. 
```
../../libraries/AP_Common/tests/test_location.cpp:112: Failure
Expected equality of these values:
  diff.y
    Which is: 1.2161366
  test.expected_pos_change.y
    Which is: 1.2050807
[  FAILED  ] Location.LocOffsetDouble (0 ms)
```

File: libraries/AP_Math/examples/location/location.cpp [hunk 1]
```diff
@@ -259,16 +257,16 @@ static void test_wgs_conversion_functions(void)
...
-    llhs[2] = Vector3d(0, 90*D2R, 0);   /* On the Equator. */
-    llhs[3] = Vector3d(0, -90*D2R, 0);  /* On the Equator. */
-    llhs[4] = Vector3d(90*D2R, 0, 0);   /* North pole. */
-    llhs[5] = Vector3d(-90*D2R, 0, 0);  /* South pole. */
-    llhs[6] = Vector3d(90*D2R, 0, 22);  /* 22m above the north pole. */
-    llhs[7] = Vector3d(-90*D2R, 0, 22); /* 22m above the south pole. */
-    llhs[8] = Vector3d(0, 0, 22);       /* 22m above the Equator and Prime Meridian. */
-    llhs[9] = Vector3d(0, 180*D2R, 22); /* 22m above the Equator. */
+    llhs[0] = Vector3d(0, 0, 0);           /* On the Equator and Prime Meridian. */
+    llhs[1] = Vector3d(0, M_PI_2, 0);     /* On the Equator. */
```
Comment (khancyr) [inline libraries/AP_Math/examples/location/location.cpp:261]: M_2PI ? 

180 * M_PI_DOUBLE / 90 = M_2PI  normally ?

**Pull request closed without merging**